lib/ansible/parsing/mod_args.py

```diff
@@ -301,6 +301,11 @@ def parse(self, skip_action_validation=False):
                     raise AnsibleParserError("conflicting action statements: %s, %s" % (action, item), obj=self._task_ds)
                 action = item
                 thing = value
+
+                context = module_loader.find_plugin_with_context(item, check_aliases=True, collection_list=self._collection_list)
```

Comment:
This doesn't cover actions and breaks `skip_action_validation=True` in some cases. It's doing the right thing, but you can deconstruct the `has_plugin` calls from conditional above into `find_plugin_with_context` calls instead, then hang onto the result and use it here.